File: references/references.py
class Person:
    def __init__(self, numbers):
        self.favorite_numbers = numbers

default_numbers = [42]
william = Person(default_numbers)
william.favorite_numbers.append(1066)
# OOPS: William and George are both referencing the same list.
george = Person(default_numbers)
george.favorite_numbers.append(1776)
print("william:", william.favorite_numbers);
print("george:", george.favorite_numbers);

File: references/references.rs
pub struct Person {
    pub favorite_numbers: Vec<i32>,
}

fn main() {
    let default_numbers = vec![42];

    let mut william = Person {
        favorite_numbers: default_numbers.clone(),
    };
    william.favorite_numbers.push(1066);

    let mut george = Person {
        favorite_numbers: default_numbers,
    };
    george.favorite_numbers.push(1776);

    println!("william: {:?}", william.favorite_numbers);
    println!("george: {:?}", george.favorite_numbers);
}
